File: Modules/SMES/MES/SIE.Web.MES/LoadItems/DeductItems/Commands/WoCostItemSubmitCommand.cs
using SIE.MES.LoadItems;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Text;

namespace SIE.Web.MES.LoadItems.DeductItems.Commands
{
    /// <summary>
    /// 工单耗用单提交命令
    /// </summary>
    public class WoCostItemSubmitCommand : ViewCommand
    {
        /// <summary>
        /// 执行
        /// </summary>
        /// <param name="args"></param>
        /// <param name="scope"></param>
        /// <returns></returns>
        /// <exception cref="NotImplementedException"></exception>
        protected override object Excute(ViewArgs args, string scope)
        {
            var data = args.Data.ToJsonObject<WoCostItem>();
            RT.Service.Resolve<WoCostItemController>().SubmitCost(data);
            return true;
        }
    }
}

File: Modules/SMES/MES/SIE.Web.MES/LoadItems/DeductItems/Commands/WoCostItemSubmitCommand.js
SIE.defineCommand("SIE.Web.MES.LoadItems.DeductItems.Commands.WoCostItemSubmitCommand", {
    meta: { text: "提交", group: "edit", iconCls: "icon-Check icon-blue" },
    canExecute: function (view) {
        var entity = view.getCurrent();
        if (entity != null && entity.isNew()) {
            return false;
        }
        //扣料状态 10 待提交
        if (entity == null || entity.getState() !== 10 || entity.getRecordType() === 30) {
            return false;
        }
        
        return true;
    },
    execute: function (view) {
        var data = view.getCurrent().data;
        SIE.Msg.wait("正在提交......".t());
        view.execute({
            data: data,
            success: function (res) {
                SIE.Msg.showMessage("提交成功!".t());
                view.reloadData();
            }
        });
    }
});
